src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -107,16 +110,24 @@ class BlockingProcess(
     }
 
     try {
-      val stdin =
-        context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
-
-      val stdout =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getInputStream, stdioTimeout), "stdout"))
-
-      val stderr =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getErrorStream, stdioTimeout), "stderr"))
-
-      context.parent ! Started(ActorSubscriber(stdin), ActorPublisher(stdout), ActorPublisher(stderr))
+      val stdin = Sink.outputStream(process.getOutputStream)
+        .mapMaterializedValue { futureResult =>
+          futureResult.onComplete(_ => context.stop(self))
```

Comment:
What used to happen was if a `Destroy` message was received at `BlockingProcess` then it would call `context.stop(self)`. In addition, if any child actor stopped then it would stop.

There was a special `ProcessDestroyer` that would kill the associated process if its is stopped. The `ProcessDestroyer` is stopped when the parent `BlockingProcess` is stopped i.e. there is a `postStop` handler in the destroyer.

Killing the actual process should cause the stdio of the process to close.

So, shutting down the `BlockingProcess` at the end of the publisher isn't correct. I'd say it was because we were previously watching the stdio actors and when they terminated, the `BlockingProcess` would cause the destroyer to stop, which in turn killed the process.